bugbot/people.py

```diff
@@ -27,27 +27,54 @@
 ]
 IM_NICK = re.compile(r"([\w\.@]+)")
 
+DEFAULT_PATH = "./configs/people.json"
+
+
+_ManagerInfo = TypedDict("_ManagerInfo", {"cn": str, "dn": str})
+
+Person = TypedDict(
+    "Person",
+    {
+        "bugzillaEmail": str,
+        "bugzillaID": str,
+        "cn": str,
+        "dn": str,
+        "found_on_bugzilla": bool,
+        "im": list[str],
+        "isdirector": str,
+        "ismanager": str,
+        "mail": str,
+        "manager": _ManagerInfo,
+        "title": str,
+    },
+)
+
 
 class People:
     _instance = None
 
-    def __init__(self, p=None):
-        if p is None:
-            with open("./configs/people.json", "r") as In:
-                self.data = json.load(In)
+    def __init__(self, people_file: str | list[Person] = DEFAULT_PATH):
+        """Constructor
+
+        Args:
+            people_file: path to the people file or loaded people data.
+        """
+        if isinstance(people_file, str):
+            with open(people_file, "r", encoding="utf-8") as file:
```

Comment:
This suggestion is not necessary (due to the assumption that `DEFAULT_PATH` exists), but maybe we could consider adding error handling for file access and JSON parsing?

Reply:
If the file is not exited, it will crash, which will indicate that it is missing. There is not much benefit of adding error handling here since we cannot proceed if there is any error.